docs/amber.md

```diff
@@ -0,0 +1,138 @@
+# Amber
+
+Amber is a multi-API shader test framework.Graphics bugs can be captured and
+communicated through a scripting interface. This removes the need to program
+to the low level interface when reproducing bugs.
+
+Amber is broken into multiple parts, the applications, the parsing components
+and the script execution.
+
+## Applications
+There are currently two applications, the `samples/amber.cc` amber application
+and the amber integration into the [CTS](https://github.com/KhronosGroup/VK-GL-CTS/tree/master/external/vulkancts/modules/vulkan/amber). These applications are responsible
+for configuring the script execution environment (setting up Vulkan, Dawn or
+another engine), calling into the parsing code to generate a test script and
+then passing that script into the script execution component.
+
+We require the application to configure the execution engine. This allows the
+application to handle loading function pointers, doing configuration and other
+setups as required. There are no hardcoded assumptions in amber as to how you're
+setting up the API to be tested.
+
+This engine configuration is done through the `VulkanEngineConfig` in
+`amber/amber_vulkan.h` and the `DawnEngineConfig` in `amber/amber_dawn.h`.
+
+The sample application does this configuration through the config_helper
+classes. `samples/config_helper_vulkan.*` and `samples/config_helper_dawn.*`.
+For the CTS, the Vulkan engine is already configured and we just set the
+`VulkanEngineConfig` as needed.
+
+Accessing amber itself is done through the `Amber` class in `amber/amber.h`. We
+require the application to parse and execute a script as two separate steps.
+This separation allows for the scripts to be retrieved after a parse command
```

Comment:
Instead of scripts, do you mean shaders?

Reply:
Yes, yes I do.